src/wallet/rpcwallet.cpp

```diff
@@ -296,11 +303,11 @@ static UniValue getrawchangeaddress(const JSONRPCRequest& request)
         pwallet->TopUpKeyPool();
     }
 
-    OutputType output_type = pwallet->m_default_change_type != OutputType::CHANGE_AUTO ? pwallet->m_default_change_type : pwallet->m_default_address_type;
-    if (!request.params[0].isNull()) {
-        if (!ParseOutputType(request.params[0].get_str(), output_type)) {
-            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, strprintf("Unknown address type '%s'", request.params[0].get_str()));
-        }
+    ColorIdentifier colorId;
+    if (!request.params[0].isNull())
+    {
+        const std::vector<unsigned char> vColorId(ParseHex(request.params[0].get_str()));
+        colorId = ColorIdentifier(vColorId);
```

Comment:
Then ColorIdentifier constructor will detec it and `TokenTypes::NONE` will be set. So the returned address will not be coloured.

Reply:
I think it would be better to make it an error if the returned address is not a Colored Address when color id is specified as an input. This way, user can avoid mistakenly assuming that the returned value is a Colored Address.

Reply:
@Naviabheeman  4d24117 check is also needed here(`getrawchangeaddress`).